examples/osm-vector-tiles.js

```diff
@@ -0,0 +1,139 @@
+goog.require('ol.Map');
+goog.require('ol.View');
+goog.require('ol.format.TopoJSON');
+goog.require('ol.layer.VectorTile');
+goog.require('ol.proj');
+goog.require('ol.source.VectorTile');
+goog.require('ol.style.Fill');
+goog.require('ol.style.Stroke');
+goog.require('ol.style.Style');
+
+
+var roadStyleCache = {};
+var landuseStyleCache = {};
+var buildingStyle = [
+  new ol.style.Style({
+    fill: new ol.style.Fill({
+      color: '#666',
+      opacity: 0.4
+    }),
+    stroke: new ol.style.Stroke({
+      color: '#444',
+      width: 1
+    })
+  })
+];
+
+var map = new ol.Map({
+  layers: [
+    new ol.layer.VectorTile({
+      source: new ol.source.VectorTile({
+        format: new ol.format.TopoJSON(),
+        tileGrid: ol.tilegrid.createXYZ({ maxZoom: 19 }),
+        url: 'http://{a-c}.tile.openstreetmap.us/' +
+            'vectiles-water-areas/{z}/{x}/{y}.topojson'
+      }),
+      style: new ol.style.Style({
+        fill: new ol.style.Fill({
+          color: '#9db9e8'
+        })
+      })
+    }),
+    new ol.layer.VectorTile({
+      source: new ol.source.VectorTile({
+        format: new ol.format.TopoJSON(),
+        tileGrid: ol.tilegrid.createXYZ({ maxZoom: 19 }),
+        url: 'http://{a-c}.tile.openstreetmap.us/' +
+            'vectiles-highroad/{z}/{x}/{y}.topojson'
+      }),
+      style: function(feature, resolution) {
+        var kind = feature.get('kind');
+        var railway = feature.get('railway');
+        var sort_key = feature.get('sort_key');
+        var styleKey = kind + '/' + railway + '/' + sort_key;
+        var styleArray = roadStyleCache[styleKey];
+        if (!styleArray) {
+          var color, width;
+          if (railway) {
+            color = '#7de';
+            width = 1;
+          } else {
+            color = {
+              'major_road': '#776',
+              'minor_road': '#ccb',
+              'highway': '#f39'
+            }[kind];
+            width = kind == 'highway' ? 1.5 : 1;
+          }
+          styleArray = [new ol.style.Style({
```

Comment:
no need for an array anymore given Tim's recent change to allow returning style objects?